bench: refactor random number generation in `stats/base/dists/negative-binomial`

yuvimittal opened the pull request:
Resolves #4980 

## Description

> What is the purpose of this pull request?

This pull request:

-  Refactor random number generation in JS benchmarks

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #4980 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

yuvimittal committed:
feat : Refactor random number generation in JS benchmarks

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/cdf/benchmark/benchmark.js:
```diff
@@ -32,18 +33,26 @@ var cdf = require( './../lib' );
+	r = new Float64Array( len );
+	p = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = uniform( 0, 100.0 );
```
```suggestion
		x[ i ] = uniform( 0.0, 100.0 );
```


Applies throughout the entire PR.

anandkaranubc replied:
This change is still needed.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/cdf/benchmark/benchmark.js:
```diff
@@ -32,18 +33,26 @@ var cdf = require( './../lib' );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = uniform( 0, 100.0 );
+		r[ i ] = discreteUniform( 1, 100 );
+		p[ i ] = uniform( EPS, 1 );
```
```suggestion
		p[ i ] = uniform( EPS, 1.0 );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/variance/benchmark/benchmark.js:
```diff
@@ -32,16 +33,23 @@ var variance = require( './../lib' );
+	r = new Float64Array( len );
+	p = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		r[ i ] = discreteUniform( 1.0, 100.0 );
```
```suggestion
		r[ i ] = discreteUniform( 1, 100 );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/variance/benchmark/benchmark.js:
```diff
@@ -32,16 +33,23 @@ var variance = require( './../lib' );
-		r = ceil( randu()*100.0 );
-		p = ( randu()*1.0 ) + EPS;
-		y = variance( r, p );
+		y = variance( r[i % len], p[i % len] );
```
```suggestion
		y = variance( r[ i % len ], p[ i % len ] );
```

anandkaranubc commented:
Thanks, @yuvi-mittal, for working on this. I left a few comments for you to review. These suggestions are applicable throughout the PR.

Also, can you change the title to:

bench: refactor random number generation in `stats/base/dists/negative-binomial`

yuvimittal committed:
linting errors

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/cdf/benchmark/benchmark.js:
```diff
@@ -32,18 +33,26 @@ var cdf = require( './../lib' );
+	p = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = uniform( 0, 100.0 );
+		r[ i ] = discreteUniform( 1.0, 100 );
```
```suggestion
		r[ i ] = discreteUniform( 1, 100 );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/ctor/benchmark/benchmark.js:
```diff
@@ -33,15 +34,22 @@ var NegativeBinomial = require( './../lib' );
+	r = new Float64Array( len );
+	p = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		r[ i ] = discreteUniform( 1.0, 50 );
```
```suggestion
		r[ i ] = discreteUniform( 1, 50 );
```

This change is consistent throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/ctor/benchmark/benchmark.js:
```diff
@@ -90,12 +99,16 @@ bench( pkg+'::set:r', function benchmark( b ) {
+	len = 100;
+	y = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		y[ i ] = discreteUniform( 1.0, 100.0 );
```
```suggestion
		y[ i ] = discreteUniform( 1, 50 );
```

Make sure the range is still the same for consistency.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/ctor/benchmark/benchmark.js:
```diff
@@ -143,12 +157,16 @@ bench( pkg+'::set:p', function benchmark( b ) {
-		y = randu() + EPS;
-		dist.p = y;
-		if ( dist.p !== y ) {
+		dist.p = y[i % len];
+		if ( dist.p !== y[i % len] ) {
```
```suggestion
		dist.p = y[ i % len ];
		if ( dist.p !== y[ i % len ] ) {
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/negative-binomial/ctor/benchmark/benchmark.js:
```diff
@@ -170,10 +190,15 @@ bench( pkg+':kurtosis', function benchmark( b ) {
 	b.tic();
 	for ( i = 0; i < b.iterations; i++ ) {
-		dist.r = ceil( ( 100.0*randu() ) + EPS );
+		dist.r = x[i % 100];
```
```suggestion
		dist.r = x[ i % len ];
```

anandkaranubc replied:
This change is consistent throughout the PR.
